recovery/elliptics_recovery/types/merge.py

```diff
@@ -686,6 +683,130 @@ def succeeded(self):
         return self.result
 
 
+class ServerSendRecovery(object):
+    '''
+    Special recovery class that tries to recover keys from backends that
+    should not contain this keys to proper backend via server-send operation.
+    '''
+    def __init__(self, ctx, node, group):
+        self.routes = self._prepare_routes(ctx, group)
+        self.session = elliptics.Session(node)
+        self.session.exceptions_policy = elliptics.exceptions_policy.no_exceptions
+        self.session.set_filter(elliptics.filters.all)
+        self.session.timeout = 60
+        self.session.groups = [group]
+        self.session.trace_id = ctx.trace_id
+        self.ctx = ctx
+
+    def _prepare_routes(self, ctx, group):
```

Comment:
Is it really needed? Why not just use `routes.get_id_routes()` for filtering keys?